File: Day 25 - Running Time and Complexity/solution.js
function processData(input) {
    //Enter your code here
    var arr = input.split('\n');
    for (var i = 1; i < arr.length; i++){
        var n = arr[i];
        if (isPrime(n)){
            console.log("Prime");
        }else {
            console.log("Not prime");
        }
    }
} 

function isPrime(n){
    if (n <= 1){
        return false;
    }
    if (n <= 3){
        return true;
    }

    if (n % 2 == 0 || n % 3 == 0){
        return false;
    }

    return true;
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});

File: Day 25 - Running Time and Complexity/Solution.java
import java.util.Scanner;

public class Solution {

    public static boolean isPrime(int n){
        for (int i = 2; i <= Math.sqrt(n); i++){
            if(n % i == 0){
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner input = new Scanner(System.in);
        int T = input.nextInt();

        for (int i = 0; i < T; i++){
            int n = input.nextInt();

            if(n >= 2 && isPrime(n))
                System.out.println("Prime");
            else System.out.println("Not prime");
        }
        input.close();
    }
}